pkg/controller/traffic/utils_test.go

```diff
@@ -0,0 +1,104 @@
+package traffic
+
+import (
+	"fmt"
+
+	shipper "github.com/bookingcom/shipper/pkg/apis/shipper/v1alpha1"
+	shipperfake "github.com/bookingcom/shipper/pkg/client/clientset/versioned/fake"
+	shipperinformers "github.com/bookingcom/shipper/pkg/client/informers/externalversions"
+	shippertesting "github.com/bookingcom/shipper/pkg/testing"
+	corev1 "k8s.io/api/core/v1"
+	kubefake "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/record"
+)
+
+// NOTE: this does not try to implement endpoint subsets at all. All
+// pods are assumed to be part of the same subset and to be always ready.
+func shiftPodInEndpoints(pod *corev1.Pod, endpoints *corev1.Endpoints) *corev1.Endpoints {
+	var podGetsTraffic bool
+	trafficStatusLabel, ok := pod.Labels[shipper.PodTrafficStatusLabel]
+	if ok {
+		podGetsTraffic = trafficStatusLabel == shipper.Enabled
+	}
+
+	addresses := endpoints.Subsets[0].Addresses
+	addressIndex := -1
+	for i, address := range addresses {
+		if address.TargetRef.Name == pod.Name {
+			addressIndex = i
+			break
+		}
+	}
+
+	if podGetsTraffic && addressIndex == -1 {
+		address := corev1.EndpointAddress{
+			TargetRef: &corev1.ObjectReference{
+				Kind:      "Pod",
+				Namespace: pod.Namespace,
+				Name:      pod.Name,
+			},
+		}
+
+		endpoints.Subsets[0] = corev1.EndpointSubset{
+			Addresses: append(addresses, address),
+		}
```

Comment:
Are we guaranteed to never loose any additional attributes from `endpoints.Subsets[0]`? What i mean is: this structure is being re-created with the only attribute `Addresses`. Any chance it would be existing with some other data that might be accidentally lost?